git.adoc

```diff
@@ -0,0 +1,211 @@
+
+. The master branch must be always deployable, i.e. fully working.
+. Do not commit directly to master, create a topic branch instead.
+. Perform code review and run all tests *before* merging into master.
+. Never do a force push into the master branch on the server.
+
+
+=== Release branches
+
+A release branch is created from master with each released _minor version_ `X.Y`.
+Recommended naming scheme is `stable/X.Y` or `X.Y-stable`. TODO ?
+
+Release branch is a long-lived branch used for porting bug fixes to the stable release `X.Y.z`.
+If possible, first merge the bug fixes into the master and then cherry-pick them into the release branch.
+Every time a bug fix is included, increment the _patch version_ (e.g. `X.Y.0` -> `X.Y.1`) and release it by setting a new tag.
+
+TODO
+
+
+=== Topic branches
+
+A topic branch (also called a feature branch) is a short-lived branch you create and use for a single particular feature, bug fix or whatever you’re working on.
+
+. Create a topic branch from the current HEAD of the master branch.
+. Regularly push your work to the branch of the same name on the server.
+. Feel free to rebase and force push into this branch until you open a pull request.
+. When the branch is ready for code review, or you need feedback or help, open a pull request.
+. Clean up the branch before merging into the master.
++
+TODO
+
+
+=== How to…
+
+==== Implement new feature or fix
+
+.Implementation phase
+. Create new topic branch from the _master_.
+. Start coding. Commit each fix or task you have done and regularly push your work to the same named branch on the server.
+. Update changelog.
+. Push your changes to the server and open a pull request.
+
+[start=5]
+.Code review phase
+. Ask co-workers to review your code.
+. Resolve issues found during review, commit them and push them into the topic branch (do not use rebase now).
+. Repeat the previous two steps until all issues are resolved.
+
+[start=8]
+.Merge phase
+. Clean up your branch; use interactive rebase to squash “checkpoint” commits and fixes... TODO
+. Rebase to master... TODO ?
+. Do a force push to your branch.
+. Ask the project’s maintainer to merge your changes into master.
+
+
+==== Release new version
+
+. Bump version number in project files (like `.gemspec`, `Cargo.toml`, …) and enter release date into changelog.
+. Commit changes into the _master_ with message “Release version X.Y.Z”.
+. If this is the first release with minor version `X.Y`, create a _release branch_ named `stable/X.Y`.
+. Create a _release tag_ `vX.Y.Z`.
+. Push changes to the server.
+
+
+==== Fix bug in released version
+
+. Fix the bug (in the _master branch_); follow the procedure <<Implement new feature or fix>>.
+. Cherry-pick commit(s) with the bug fix to the corresponding _release branch_.
+. <<Release new version,Release new patch version>>, but commit and push to the _release branch_ instead of _master_.
+
+
+
+== Tips
```

Comment:
add +x howto/tip for executable files stored on filesystem without mode attributes